extensions-core/druid-basic-security/src/main/java/io/druid/security/basic/BasicAuthUtils.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.security.basic;
+
+import io.druid.java.util.common.StringUtils;
+import io.druid.java.util.common.logger.Logger;
+
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+import javax.servlet.http.HttpServletRequest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Base64;
+
+public class BasicAuthUtils
+{
+  private static final Logger log = new Logger(BasicAuthUtils.class);
+  private static final Base64.Encoder ENCODER = Base64.getEncoder();
+  private static final Base64.Decoder DECODER = Base64.getDecoder();
+  private static final SecureRandom SECURE_RANDOM = new SecureRandom();
+  public final static String ADMIN_NAME = "admin";
+  public final static String INTERNAL_USER_NAME = "druid_system";
+
+  public static int SALT_LENGTH = 32;
+  public static int KEY_ITERATIONS = 10000;
```

Comment:
I changed the iteration counts so that the default value is a property on each Authenticator, and there's also an option to specify an iteration count on the credential update API. This makes it possible to update the iteration count by resetting the password using the API.